pkg/sync/sync_file.go

```diff
@@ -812,6 +817,16 @@ func (sf *SyncingFile) updateSyncReadyNoLock() {
 	})
 }
 
+func (sf *SyncingFile) updateVirtualSizeNoLock(filePath string) {
+	// This only works if filePath is valid - sometimes we need to call it
+	// with sf.tmpFilePath, sometimes with sf.filePath :-/
+	virtualSize, err := util.GetImageVirtualSize(filePath)
+	if err != nil {
+		sf.log.Warnf("SyncingFile: failed to get backing image virtual size: %v", err)
+	}
+	sf.virtualSize = virtualSize // This will be zero in case of error.  Not sure if that's what we want or not.
```

Comment:
I am fine with leaving it 0 or falling back to use the actual size.

Reply:
I prefer to leave it as zero in case of error rather than falling back to the physical image size, because it's the only way that components further up the stack could know that the virtualSize somehow isn't available yet.

Reply:
(should I remove that comment?)

Reply:
You can leave a comment like `when there is an error, leaving `sf.virtualSize` to be zero so that components further up the stack could know that the virtual size somehow isn't available yet`

Reply:
Done!